app/webpacker/controllers/guest_checkout_controller.js

```diff
@@ -34,4 +38,21 @@ export default class extends Controller {
   usingGuestCheckout() {
     return sessionStorage.getItem(this.sessionValue) === this.distributorValue;
   }
+
+  handlePageUnload(event) {
+    event.preventDefault()
+
+    event.returnValue = 'I18n.t("admin.unsaved_confirm_leave")';
+    return 
+  }
+
+  removeUnloadEvent() {
+    if (this.hasSummaryTarget) {
+      window.removeEventListener('beforeunload', this.handlePageUnload);
+    }
+  }
+
+  disconnect() {
```

Comment:
This is the Stimulus lifecycle method and it's good practice to write Stimulus-specific methods/properties at the beginning of the controller before writing any controller actions. This way it's easier to maintain and differentiate between methods.
It would be great if we could move it next to the `connect` lifecycle method.